src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * The duke.Storage class handles the loading and saving of tasks from and to a file.
+ * It is responsible for managing the persistence of the task list.
+ */
+public class Storage {
+
+
+    private ArrayList<IndividualTask> tasks = new ArrayList<IndividualTask>();
+
+    private String path;
+
+    /**
+     * Constructs a duke.Storage object with the specified file path.
+     * It automatically attempts to load tasks from the specified file.
+     *
+     * @param path The file path where tasks are stored.
+     */
+    public Storage(String path) {
+        this.path = path;
+        this.loadTasksFromFile();
+    }
+
+    /**
+     * Loads tasks from the specified file into the task list.
+     * If the directory or file does not exist, it provides instructions for creating them.
+     * The tasks are read from the file and converted into duke.IndividualTask objects,
+     * which are added to the task list.
+     */
+    public void loadTasksFromFile() {
+        File directory = new File("./data");
+        File file = new File(path);
+        if (!directory.exists()) {
+
+            System.out.println("No existing directory found. "
+                    + "Create a directory data and a file 'duke.txt' inside that directory. "
+                    + "Starting with an empty task list.");
+
+            return;
+        }
+        if (!file.exists()) {
+            System.out.println("No existing data file found. "
+                    + "Create a file 'duke.txt' in the data directory. "
+                    + "Starting with an empty task list.");
+            return;
+        }
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String tasks = scanner.nextLine();
+                String[] parts = tasks.split(" \\| ");
+                String typeOfTask = parts[0];
+                boolean isTaskDone = parts[1].equals("1");
+                String descriptionOfTask = parts[2];
+                IndividualTask task = switch (typeOfTask) {
```

Comment:
Personally, I would use the switch as follows:

`
IndividualTask task = null;

switch (typeOfTask) {
case "T":
task = new Todo (descriptionOfTask);
...
`